src/charm.py

```diff
@@ -110,335 +202,226 @@ def set_pod_spec(self, event):
+        """Upload generated certs to container."""
+        self.container.push(
+            "/tmp/k8s-webhook-server/serving-certs/tls.key",
+            self._stored.key,
+            make_dirs=True,
+        )
+        self.container.push(
+            "/tmp/k8s-webhook-server/serving-certs/tls.crt",
+            self._stored.cert,
+            make_dirs=True,
+        )
+        self.container.push(
+            "/tmp/k8s-webhook-server/serving-certs/ca.crt",
+            self._stored.ca,
+            make_dirs=True,
+        )
 
-    def __init__(self, msg, status_type=None):
-        super().__init__()
+    def _deploy_k8s_resources(self) -> None:
+        """Deploys K8S resources."""
+        try:
+            self.unit.status = MaintenanceStatus("Creating K8S resources")
+            self.k8s_resource_handler.apply()
+            self.crd_resource_handler.apply()
+            self.configmap_resource_handler.apply()
+        except ApiError:
+            raise ErrorWithStatus("K8S resources creation failed", BlockedStatus)
+        self.model.unit.status = MaintenanceStatus("K8S resources created")
+
+    def _on_install(self, _):
+        """Perform installation only actions."""
+        # upload certs to container
+        self._check_container_connection()
+        self._upload_certs_to_container()
+
+        # proceed with other actions
+        self.main(_)
+
+    def _on_remove(self, _):
+        """Remove all resources."""
+        self.unit.status = MaintenanceStatus("Removing K8S resources")
+        k8s_resources_manifests = self.k8s_resource_handler.render_manifests()
+        crd_resources_manifests = self.crd_resource_handler.render_manifests()
+        configmap_resources_manifests = self.configmap_resource_handler.render_manifests()
+        try:
+            delete_many(self.crd_resource_handler.lightkube_client, crd_resources_manifests)
+            delete_many(self.k8s_resource_handler.lightkube_client, k8s_resources_manifests)
+            delete_many(
+                self.configmap_resource_handler.lightkube_client,
+                configmap_resources_manifests,
+            )
+        except ApiError as e:
+            self.logger.warning(f"Failed to delete K8S resources, with error: {e}")
+            raise e
+        self.unit.status = MaintenanceStatus("K8S resources removed")
 
-        self.msg = str(msg)
-        self.status_type = status_type
-        self.status = status_type(self.msg)
+    def _gen_certs(self):
+        """Generate certificates."""
+        # generate SSL configuration based on template
+        model = self.model.name
+
+        try:
+            ssl_conf_template = open(SSL_CONFIG_FILE)
+            ssl_conf = ssl_conf_template.read()
+        except ApiError as error:
+            self.logger.warning(f"Failed to open SSL config file: {error}")
+
+        ssl_conf = ssl_conf.replace("{{ model }}", str(model))
+        Path("/tmp/seldon-cert-gen-ssl.conf").write_text(ssl_conf)
```

Comment:
Can we move all the `/tmp/seldon*` to instead use a temp directory generated by python?  That way we cannot get any crosstalk between two tests or seldon instances

Reply:
Which directory?

Reply:
Done.